openshift-client/src/main/java/io/fabric8/openshift/client/OpenShiftConfig.java

```diff
@@ -55,6 +58,14 @@ public OpenShiftConfig(Config kubernetesConfig) {
     );
   }
 
+  public OpenShiftConfig(Config kubernetesConfig, String apiGroupUrl, String apiGroupVersion) {
+    this(kubernetesConfig,
+      getDefaultOpenShiftUrl(kubernetesConfig), getDefaultOapiVersion(kubernetesConfig), DEFAULT_BUILD_TIMEOUT
+    );
+    this.openShiftUrl = apiGroupUrl;
```

Comment:
ApiGroupUrl and apiGroupVersion are not exaclty user configuration and should not be handled here.
Also using the openshiftUrl as carrier for this, is dangerous and will cause issues.

The only thing that does make sense to be here is `disableApiGroupCheck`. Everything else should move into the implementation classes.

Each implementation class has a methods like `getResourceUrl()` and `getNamesacedUrl()` I would override them so that they handle the logic there. `OpenShiftOperation` might be good place for this.